api/src/adapters/search/opensearch_client.py

```diff
@@ -4,9 +4,34 @@
 import opensearchpy
 
 from src.adapters.search.opensearch_config import OpensearchConfig, get_opensearch_config
+from src.adapters.search.opensearch_response import SearchResponse
 
 logger = logging.getLogger(__name__)
 
+# By default, we'll override the default analyzer+tokenization
+# for a search index. You can provide your own when calling create_index
+DEFAULT_INDEX_ANALYSIS = {
+    "analyzer": {
+        "default": {
+            "type": "custom",
+            "filter": ["lowercase", "custom_stemmer"],
+            # Change tokenization to whitespace as the default is very clunky
+            # with a lot of our IDs that have dashes in them.
+            # see: https://opensearch.org/docs/latest/analyzers/tokenizers/index/
+            "tokenizer": "whitespace",
+        }
+    },
+    # Change the default stemming to use snowball which handles plural
+    # queries better than the default
+    # TODO - there are a lot of stemmers, we should take some time to figure out
+    #        which one works best with our particular dataset. Snowball is really
+    #        basic and naive (literally just adjusting suffixes on words in common patterns)
+    #        which might be fine generally, but we work with a lot of acronyms
+    #        and should verify that doesn't cause any issues.
+    # see: https://opensearch.org/docs/latest/analyzers/token-filters/index/
+    "filter": {"custom_stemmer": {"type": "snowball", "name": "english"}},
```

Comment:
This looks like a good default to start with. At some point there should be a content strategy or the like for the indexing options. For acronyms we can also create a huge list of synonyms.

Reply:
Yeah, the exact configuration we'll want is likely something we will sort out with testing. Just got this one from some basic testing of opportunity titles. Realized I needed something more than the default when my tests were getting a difference between searching `King` and `Kings`.